python/pyspark/sql/tests.py

```diff
@@ -429,6 +429,11 @@ def test_udf_with_input_file_name(self):
         row = self.spark.read.json(filePath).select(sourceFile(input_file_name())).first()
         self.assertTrue(row[0].find("people1.json") != -1)
 
+    def test_udf_should_validate_input_args(self):
+        from pyspark.sql.functions import udf
+
+        self.assertRaises(TypeError, udf(lambda x: x), None)
```

Comment:
Sounds good. It's probably worth exploring eventually, but there's no need to hold up this PR.